src/llvm/mod.rs

```diff
@@ -627,6 +627,47 @@ impl<'g> CodeGen<'g> for hir::If {
     }
 }
 
+impl<'g> CodeGen<'g> for hir::Else {
+    fn codegen(&self, generator: &mut Generator<'g>) -> BasicValueEnum<'g> {
+        let current_function = generator.current_function();
+        let expr_block = generator.context.append_basic_block(current_function, "expr");
+        let end_block = generator.context.append_basic_block(current_function, "end_else");
+
+        // Setup conditional jump
+        let else_block = generator.context.append_basic_block(current_function, "else");
```

Comment:
We don't need a new block for `expr` - it can be evaluated in the current block since we don't need to conditionally jump to it or anything.

You can think of `a else b` as desugaring to:
```hs
match a
| Some value -> value
| None -> b
```
Which in llvm is something like the following pseudocode:
```llvm
%0 = ... evaluate a ...
%1 = extract tag value from %0
br i1 %1, label %none_case, label %some_case              (build_conditional_branch)

none_case:
  %2 = ... evaluate b ...
  br label %end
some_case:
  %3 = extract non-tag value from %0
  br label %end
end:
  %4 = phi <result type> [%2, %none_case ], [%3, %some_case]                    result of the match
```

You can always use `ante --emit ir --backend llvm <filename>` to see what llvm you're emitting for a particular file

Reply:
That makes sense! One thing I'm trying to learn is how exactly to extract out the tag and non-tag values from the codegen'd IR? Is there a specific section of the llvm codegen I could potentially pattern match off?

Reply:
Yeah, you can look at the codegen portions for decision trees which is what match expressions are compiled to. It can be a bit tricky though since it depends on type information. Another option would be desugaring the Else nodes into Match nodes during monomorphisation most likely. Then the `hir` wouldn't need Else nodes at all.